### PR TITLE
 Modified `superscript` and `subscript` text styles to be mutually exclusive.

### LINKED ISSUE #20000: Superscript and subscript can be active at the same time (formatting conflict)
In the text formatting (e.g. RTE/editor), “Superscript” and “Subscript” can be active at the same time.
Enabling one function does not disable the other, although only one of these two formatting options should be applied to a given character/selection at a time.

## 📝 Provide detailed reproduction steps (if any)

Type some text  and select part of the text.
Click on “Superscript”.
Result: The selected text becomes superscript.
Without changing the selection, additionally click on “Subscript”.

### ✔️ Expected result

The new choice (Subscript) replaces the previous one (Superscript). The text is only subscript, not superscript anymore.
### ❌ Actual result

Both buttons can be active at the same time, leading to inconsistent formatting and potentially incorrect rendering in the editor/output.

### ❓ Possible solution

Perhaps a toogle could or a select field is possible

### PR DESCRIPTION
**Pull request by martnpaneq**
### 🚀 Summary

The `superscript` and `subscript` text styles are now mutually exclusive: applying one to text that already has the other replaces it, matching the behavior of common word processors. The previous behavior, where both attributes could coexist on the same text, can be restored by setting `config.basicStyles.superscript.allowNesting` or `config.basicStyles.subscript.allowNesting` to `true`, which is useful for content such as isotope notation or tensor indices.

---

### 📌 Related issues

* Closes #20000

---

### 💡 Additional information

https://github.com/user-attachments/assets/84aae06e-1d1a-44f0-ab1d-0164fe234169

---

### 🧾 Checklists

Use the following checklists to ensure important areas were not overlooked.
This does not apply to feature-branch merges.
If an item is **not relevant** to this type of change, simply leave it unchecked.

#### Author checklist

- [x] Is the changelog entry intentionally omitted? (Changelog entry included.)
- [x] Is the change backward-compatible? (No, but it is treated as a fix. It changes the behaviour of `superscript` and `subscript` to be aligned with other solutions on the market.)
- [x] Have you considered the impact on different editor setups and core interactions? _(e.g., classic/inline/multi-root/many editors, typing, selection, paste, tables, lists, images, collaboration, pagination)_ (Should not impact other editor setups in unpredicted way.)
- [x] Has the change been manually verified in the relevant setups? (Verified in a manual test.)
- [x] Does this change affect any of the above? (Not relevant.)
- [x] Is performance impacted? (Performance not impacted.)
- [x] Is accessibility affected? (Accessibility not affected.)
- [x] Have tests been added that fail without this change (against regression)? (Appropriate tests added.)
- [x] Have the API documentation, guides, feature digest, and related feature sections been updated where needed? (Documentation updated.)
- [x] Have metadata files (ckeditor5-metadata.json) been updated if needed? (Not relevant.)
- [x] Are there any changes the team should be informed about (e.g. architectural, difficult to revert in future versions or having impact on other features)? (This change does not have a big impact.)
- [ ] Were these changes documented (in Logbook)?

#### Reviewer checklist

- [ ] PR description explains the changes and the chosen approach (especially when  performance, API, or UX is affected).
- [ ] The changelog entry is clear, user‑ or integrator-facing, and it describes any breaking changes.
- [ ] All new external dependencies have been approved and mentioned in LICENSE.md (if any).
- [ ] All human-readable, translateable strings in this PR been introduced using `t()` (if any).
- [ ] I manually verified the change (e.g., in manual tests or documentation).
- [ ] The target branch is correct.

### CODE REVIEW

File: packages/ckeditor5-basic-styles/src/augmentation.ts [hunk 1]
```diff
@@ -28,6 +32,23 @@ import type {
...
+		 */
+		superscript?: BasicStyleSuperscriptConfig;
+
+		/**
+		 * The configuration of the {@link module:basic-styles/subscript~Subscript subscript feature}.
+		 *
+		 * Read more in {@link module:basic-styles/subscriptconfig~BasicStyleSubscriptConfig}.
+		 */
+		subscript?: BasicStyleSubscriptConfig;
+	}
```
Comment (pomek) [inline packages/ckeditor5-basic-styles/src/augmentation.ts:40]: These new options appear at the _root_ level, next to `plugins`, etc. I wonder if they could be wrapped into a namespace, e.g., `basicStyles`.

File: packages/ckeditor5-basic-styles/src/mutuallyexclusiveattributecommand.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		} );
+	}
+}
+
+function _isNestingAllowed( editor: Editor ): boolean {
+	return Boolean(
+		editor.config.get( 'superscript.allowNesting' ) ||
+		editor.config.get( 'subscript.allowNesting' )
+	);
+}
```
[the remark this answers is not in the document]
Comment (martnpaneq) [reply packages/ckeditor5-basic-styles/src/mutuallyexclusiveattributecommand.ts:93]: Resolved in [3717685](https://github.com/ckeditor/ckeditor5/pull/20116/commits/371768592079a9e935e18c60b969c398fa6f7177).

File: packages/ckeditor5-basic-styles/src/subscript/subscriptcommand.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ * @module basic-styles/subscript/subscriptcommand
+ */
+
+import type { Editor } from '@ckeditor/ckeditor5-core';
+import { ModelDocumentSelection, type ModelElement, type ModelRange } from '@ckeditor/ckeditor5-engine';
+
+import { AttributeCommand } from '../attributecommand.js';
+
+const SUBSCRIPT = 'subscript';
+const SUPERSCRIPT = 'superscript';
```
Comment (jacek-cke) [inline packages/ckeditor5-basic-styles/src/subscript/subscriptcommand.ts]: Could we move these to a shared file, since they’re also declared in other files (e.g., `subscriptediting.ts`, `superscriptcommand.ts`)?

File: .changelog/20260428123311_ck_20000.md
Comment (jacek-cke) [inline .changelog/20260428123311_ck_20000.md:1]: nit: It may not be important, but I believe the changelog filename format is `YYYYMMDDHHMMSS_{prefix}_{issue}.md`.

File: packages/ckeditor5-basic-styles/src/augmentation.ts (hunk 1 above)
Comment (martnpaneq) [reply packages/ckeditor5-basic-styles/src/augmentation.ts:40]: Introduced a `basicStyles` namespace in [518e4a0](https://github.com/ckeditor/ckeditor5/pull/20116/commits/518e4a0c53782db290817751185899693cc073f6).

File: packages/ckeditor5-basic-styles/src/subscript/subscriptcommand.ts (hunk 3 above)
Comment (martnpaneq) [reply packages/ckeditor5-basic-styles/src/subscript/subscriptcommand.ts]: Extracted to a common place in [b921abc](https://github.com/ckeditor/ckeditor5/pull/20116/commits/b921abc1eb11c7d80caabee1bcf25d5c4ccaf093).

File: .changelog/20260428123311_ck_20000.md
Comment (martnpaneq) [reply .changelog/20260428123311_ck_20000.md:1]: Updated in [32a625e](https://github.com/ckeditor/ckeditor5/pull/20116/commits/32a625e79c53bdf951bad9252c69a945299d5a7d).

File: .changelog/20260428123311_ck_20000.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+type: Fix
+scope:
+  - ckeditor5-basic-styles
+---
+
+The `superscript` and `subscript` text styles are now mutually exclusive.
```
Comment (pomek) [inline .changelog/20260428123311_ck_20000.md]: It would be good to mention that this is not the default behavior. A user can configure the editor to work this way. And based on that, it results in being more of a feature than a fix, but I am not forcing the type change.

Comment (martnpaneq) [reply .changelog/20260428123311_ck_20000.md]: Actually it is the default behaviour now. If you think we should make it not default - please let me know.

Comment (martnpaneq) [reply .changelog/20260428123311_ck_20000.md]: This comment suggests it should be the default: https://github.com/ckeditor/ckeditor5/issues/20000#issuecomment-4132659783.

Comment (martnpaneq) [reply .changelog/20260428123311_ck_20000.md]: Updated the changeset content to be more descriptive in [71425c2](https://github.com/ckeditor/ckeditor5/pull/20116/commits/71425c2e08d2cba6d615b15133d7b9f9c8ccc538).

Comment (pomek) [reply .changelog/20260428123311_ck_20000.md]: Fair point, thanks.

**Review by jacek-cke — APPROVED** [review]
LGTM

**Comment by pomek** [comment]
Edge case, but what would happen when both the subscript and superscript attributes are defined in the data on the editor load?

```html
<p>
  Test: <sup><sub>2</sub></sup>.
</p>
```

I am wondering about the alternative approach:

Use a schema to control both attributes. This way, we do not have to provide additional configuration options. Instead, our docs can share a code snippet that would turn off this check.

By adding a postfixer, this approach resolves the potential issue with _invalid_ HTML when creating the editor.

One caveat: we need to decide which attribute is more important. For the following conflict: `<sup><sub>2</sub></sup>`, do we expect that editor keeps

- `<sup>2</sup>`, or
- `<sub>2</sub>`

Maybe we can solve it at the upcast level, but I am not sure.
